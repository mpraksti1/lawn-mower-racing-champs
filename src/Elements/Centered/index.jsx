import React from 'react'
import StyledDiv from './styles';

export default ({children}) => {
  return (
    <StyledDiv>
      {children}
    </StyledDiv>
  )
}
