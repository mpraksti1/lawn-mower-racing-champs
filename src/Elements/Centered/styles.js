import styled from 'styled-components';

const StyledDiv = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`

StyledDiv.displayName = 'CenteredStyled';

export default StyledDiv;