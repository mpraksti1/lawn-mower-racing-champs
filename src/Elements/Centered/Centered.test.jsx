import React from 'react';
import { shallow } from 'enzyme';
import Centered from '.';

describe('Centered component', () => {
  it('renders without crashing', () => {
    shallow(
      <Centered>
        <p>A centered thing</p>
      </Centered>,
    );
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <Centered>
        <p>A centered thing</p>
      </Centered>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
