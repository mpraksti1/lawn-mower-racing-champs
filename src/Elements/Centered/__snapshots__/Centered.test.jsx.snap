// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Centered component matches snapshot 1`] = `
<CenteredStyled>
  <p>
    A centered thing
  </p>
</CenteredStyled>
`;
